app/models/user.rb

```diff
@@ -58,6 +58,17 @@ def update_role(groups)
     roles_users_not_to_have.map(&:destroy)
   end
 
+  def archive!
+    attributes.each_key do |attribute|
+      self[attribute] = nil unless %w[deleted_at updated_at created_at provider id]
+                                         .include? attribute
+    end
+    self.name = "Gearchiveerde gebruiker #{self.id}"
+    self.deactivated = true
+    versions.destroy_all
```

Comment:
Please check that we shouldn't do this after delete. If the version that is now created on delete reveals all current attributes in teh changed attributes we still have the data

Reply:
You are correct, the data is still in the papertrail. This should also be fixed on Amber API since I copied the code from there.